src/components/twilio-voice-monitoring/public/twilio-voice-monitoring.js

```diff
@@ -106,37 +118,96 @@ class TwilioVoiceMonitoring extends HTMLElement {
     this.#callSid = undefined;
   };
 
+  #setDeviceHandlers(device) {
+    device.on('destroyed', () => {
+      this.#log('INFO[Device]', 'destroyed');
+    });
+    device.on('error', (twilioError, call) => {
+      const errorLog = {
+        callSid: call.parameters.CallSid,
+        code: twilioError.code,
+        message: twilioError.message,
+      };
+      this.#log('Error[Device]', JSON.stringify(errorLog, null, 2));
+    });
+    device.on('incoming', (call) => {
+      this.#log('INFO[Device]', `incoming call from: ${call.parameters.From}`);
+    });
+    device.on('registered', () => {
+      this.#log('INFO[Device]', 'registered');
+    });
+    device.on('registering', () => {
+      this.#log('INFO[Device]', 'registering');
+    });
+    device.on('tokenWillExpire', () => {
+      this.#log('INFO[Device]', 'refreshing token');
+    });
+    device.on('unregistered', () => {
+      this.#log('INFO[Device]', 'unregistered');
+    });
+  }
+
   #setCallHandlers(call) {
+    // https://www.twilio.com/docs/voice/sdks/javascript/twiliocall#events
     this.#call = call;
-    this.#call.on('accept', () => {
+    this.#call.on('accept', (call) => {
+      this.#handleCallLogging('call-accepted');
       this.#callSid = this.#call.parameters.CallSid;
     });
-    this.#call.on('disconnect', this.#reset);
-    this.#call.on('cancel', this.#reset);
-    this.#call.on('reject', this.#reset);
-    this.#call.on('messageReceived', (message) =>
-      this.#handleCallMessageReceived(message)
-    );
-    this.#call.on('error', (error) => {
-      // https://www.twilio.com/docs/voice/sdks/javascript/twiliocall#error-event
+    this.#call.on('cancel', () => {
+      this.#handleCallLogging('call-canceled');
+      this.#reset;
```

Comment:
```suggestion
      this.#reset();
```
Did you mean to invoke this?

Reply:
yes! good catch